wal/wal.go

```diff
@@ -707,25 +706,36 @@ func (r *segmentBufReader) Close() (err error) {
 	return err
 }
 
+// Read implements io.Reader.
 func (r *segmentBufReader) Read(b []byte) (n int, err error) {
-	if !r.more {
-		if r.cur+1 >= len(r.segs) {
-			return 0, io.EOF
-		}
-		r.cur++
-		r.off = 0
-		r.more = true
-		r.buf.Reset(r.segs[r.cur])
-	}
 	n, err = r.buf.Read(b)
 	r.off += n
+
+	// If we succeeded, or hit a non-EOF, we can stop.
 	if err != io.EOF {
```

Comment:
Maybe `if err==nil || err != io.EOF {`  ?
I was a it confused at the begining so maybe be a bit more mor e explicit to avoid confusion how come we succeeded here.

Reply:
@cstyan mentioned this too, will do.